ballerina/rag.bal

```diff
@@ -147,6 +147,10 @@ public distinct isolated class VectorKnowledgeBase {
 
 isolated function guessChunker(Document|Chunk doc) returns Chunker {
     // Guess the chunker based on the document type or mimeType in metadata
+    string? fileName = doc.metadata?.fileName;
+    if fileName is string && fileName.endsWith(".md") {
```

Comment:
shall we use the mimeType in metadata? as a fall back we can use the extension of the filename.